### PR TITLE
apps/examples/testcase/le_tc/kernel : Add libc signal testcase

### PR DESCRIPTION
**Pull request by hasw7569**
Add psignal testcase in libc signal to improve coverage

Artik053 log :

[tc_libc_signal_sigemptyset] PASS

[tc_libc_signal_sigfillset] PASS

[tc_libc_signal_sigaddset] PASS

[tc_libc_signal_sigdelset] PASS

[tc_libc_signal_sigismember] PASS

[tc_libc_signal_sigignore] PASS

[tc_libc_signal_raise] PASS

[tc_libc_signal_sigset] PASS

[tc_libc_signal_signal] PASS

User-defined signal 1 (POSIX): SIGUSR1
SIGUSR1
[tc_libc_signal_psignal] PASS

sigpause test : Before Pause
sigpause test : 2 sec After Pause
[tc_libc_signal_sigpause] PASS

IMXRT1050-EVKB log :

[tc_libc_signal_sigemptyset] PASS

[tc_libc_signal_sigfillset] PASS

[tc_libc_signal_sigaddset] PASS

[tc_libc_signal_sigdelset] PASS

[tc_libc_signal_sigismember] PASS

[tc_libc_signal_sigignore] PASS

[tc_libc_signal_raise] PASS

[tc_libc_signal_sigset] PASS

[tc_libc_signal_signal] PASS

User-defined signal 1 (POSIX): SIGUSR1
SIGUSR1
[tc_libc_signal_psignal] PASS

sigpause test : Before Pause
sigpause test : 2 sec After Pause
[tc_libc_signal_sigpause] PASS

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [5cffaaf4f4ee6f78be65738ec0ba8ab9763f5e76] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [5cffaaf4f4ee6f78be65738ec0ba8ab9763f5e76] - Code Rule Check (C++) OK.

File: apps/examples/testcase/le_tc/kernel/tc_libc_signal.c [hunk 1]
```diff
@@ -318,6 +318,35 @@ static void tc_libc_signal_raise(void)
...
+
+static void tc_libc_signal_psignal(void)
+{
+	siginfo_t siginfo;
+	siginfo.si_signo = SIG1;
+
+	psiginfo(NULL, NULL);
+	TC_ASSERT_EQ("psignal", get_errno(), EINVAL);
+
+	psiginfo(&siginfo, "User-defined signal 1 (POSIX)");
```
Comment (jeongchanKim) [inline apps/examples/testcase/le_tc/kernel/tc_libc_signal.c]: Please remain the expected result for comment.

File: apps/examples/testcase/le_tc/kernel/tc_libc_signal.c [hunk 2]
```diff
@@ -333,6 +362,7 @@ int libc_signal_main(void)
 	tc_libc_signal_raise();
 	tc_libc_signal_sigset();
 	tc_libc_signal_signal();
+	tc_libc_signal_psignal();
```
Comment (jeongchanKim) [inline apps/examples/testcase/le_tc/kernel/tc_libc_signal.c]: Please move this to the top(alphabet order.)

**Comment by seinfra** [comment]
Target : [36d76d2f3c085abb9788884a6415231a4e5b780a] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [36d76d2f3c085abb9788884a6415231a4e5b780a] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [b80895a76f284eb253ff0e579300719bdb8b97af] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [b80895a76f284eb253ff0e579300719bdb8b97af] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [65f380bf3d62b13497f1499344ecfde2f029efd8] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [65f380bf3d62b13497f1499344ecfde2f029efd8] - Code Rule Check OK.

File: apps/examples/testcase/le_tc/kernel/tc_libc_signal.c (hunk 1 above)
Comment (hasw7569) [reply apps/examples/testcase/le_tc/kernel/tc_libc_signal.c]: modify

File: apps/examples/testcase/le_tc/kernel/tc_libc_signal.c (hunk 2 above)
Comment (hasw7569) [reply apps/examples/testcase/le_tc/kernel/tc_libc_signal.c]: modify

**Comment by seinfra** [comment]
Target : [8f3ce0063c49dd563428428c791263f7b868873c] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [8f3ce0063c49dd563428428c791263f7b868873c] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [832f328998e1416430e5535c69a69635fb361a06] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [832f328998e1416430e5535c69a69635fb361a06] - Code Rule Check OK.

File: build/configs/qemu/tc_64k/defconfig [hunk 3]
```diff
@@ -312,6 +337,7 @@ CONFIG_MAX_TASKS=16
 CONFIG_SCHED_HAVE_PARENT=y
 # CONFIG_SCHED_CHILD_STATUS is not set
 CONFIG_SCHED_WAITPID=y
+CONFIG_SIGKILL_HANDLER=y
```
Comment (yashwanth686007) [inline build/configs/qemu/tc_64k/defconfig]: Why to enable user defined sigkill handler in this case?

File: build/configs/qemu/tc_64k/defconfig [hunk 4]
```diff
@@ -783,26 +868,20 @@ CONFIG_TASH_CMDTASK_PRIORITY=100
...
-# CONFIG_HEAPINFO_USER_GROUP is not set
-# CONFIG_HEAPINFO_USER_GROUP_LIST is not set
-# CONFIG_ENABLE_IRQINFO is not set
 # CONFIG_ENABLE_KILL is not set
 # CONFIG_ENABLE_KILLALL is not set
-# CONFIG_ENABLE_PS is not set
+CONFIG_ENABLE_PS=y
 # CONFIG_ENABLE_STACKMONITOR is not set
-# CONFIG_ENABLE_UPTIME is not set
+CONFIG_ENABLE_UPTIME=y
```
Comment (yashwanth686007) [inline build/configs/qemu/tc_64k/defconfig]: Tash is disabled and why to enable ps and uptime commands?

File: build/configs/qemu/tc_64k/defconfig [hunk 5]
```diff
@@ -379,6 +405,10 @@ CONFIG_MQ_MAXMSGSIZE=32
 # CONFIG_SCHED_HPWORK is not set
 # CONFIG_SCHED_LPWORK is not set
 
+#
+# Binary Manager
```
Comment (sunghan-chang) [inline build/configs/qemu/tc_64k/defconfig]: There is no binary manager in master branch. I think you get this from protection branch.
Please update this in master.

File: build/configs/qemu/tc_64k/defconfig (hunk 4 above)
Comment (sunghan-chang) [reply build/configs/qemu/tc_64k/defconfig]: As @yashwanth686007 said, let's disable them by default.

**Comment by seinfra** [comment]
Target : [19fa141777827ecffde7dd779e54e51e303b96a0] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [19fa141777827ecffde7dd779e54e51e303b96a0] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [4ed6579ada8ab260fdbc9d7a109b365a1ac5c22f] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [4ed6579ada8ab260fdbc9d7a109b365a1ac5c22f] - Code Rule Check OK.

File: build/configs/qemu/tc_64k/defconfig (hunk 5 above)
Comment (hasw7569) [reply build/configs/qemu/tc_64k/defconfig]: modify

File: build/configs/qemu/tc_64k/defconfig (hunk 4 above)
Comment (hasw7569) [reply build/configs/qemu/tc_64k/defconfig]: modify

File: build/configs/qemu/tc_64k/defconfig [hunk 6]
```diff
@@ -827,3 +890,8 @@ CONFIG_ENABLE_FREE=y
 # IoTBus Framework
 #
 # CONFIG_IOTBUS is not set
+
+#
+# Security Framework
+#
+# CONFIG_SECURITY_API is not set
```
Comment (sunghan-chang) [inline build/configs/qemu/tc_64k/defconfig]: Could you add newline?

**Comment by seinfra** [comment]
Target : [9bc03c5d95c48c72fa1d3346db324974f3cbb3d8] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [9bc03c5d95c48c72fa1d3346db324974f3cbb3d8] - Code Rule Check OK.

**Pull request merged**